app/models/prison.rb

```diff
@@ -3,6 +3,12 @@
 class Prison
   attr_reader :code
 
+  class << self
+    def active
+      PrisonEmumerator.new Allocation.distinct.pluck(:prison)
+    end
+  end
+
```

Comment:
@starswan
So... PrisonEnumerators job is to yield prisons. It needs the prisons codes to instantiate each prison.  Inside the active method the PrisonEnumerators is being passed the Allocation class, which has selected all allocated prisons and thereby returning only active prisons. By passing Allocations :prison inside the Prison class, is it actually passing self, is that it is rapped inside the class << self. 
PrisonEnumerator instantiates with prison codes but we are passing it prisons. I am not sure I understand how that part works.

Reply:
And the main reason for this refactor is that you want Prison to be responsible for knowing which Prisons are active, not Allocation. So it is just a transfer of responsibility. But Prison needs to talk to the Allocation model to find this out?